lib/dynamoid/transaction/create.rb

```diff
@@ -0,0 +1,37 @@
+require_relative 'action'
+
+module Dynamoid
+  module Transaction
+    class Create < Action
+
+      def run_callbacks
+        (yield; return) unless model
+        model.run_callbacks(:save) do
+          model.run_callbacks(:create) do
+            model.run_callbacks(:validate) do
+              yield
+            end
+          end
+        end
+      end
```

Comment:
As a user I would expect that `before` callback are called before persisting and `after` callback after it. So in `after` callbacks model persistence is guarantied.

Reply:
I believe since these are nested all of the before callbacks will be run first, then persisted, then after callbacks are run. Am I incorrect?

Reply:
👍